src/passes/Asyncify.cpp

```diff
@@ -503,7 +504,8 @@ class ModuleAnalyzer {
     : public ModuleUtils::CallGraphPropertyAnalysis<Info>::FunctionInfo {
     // The function name.
     Name name;
-    // If this function can start an unwind/rewind.
+    // If this function can start an unwind/rewind and not the bottommost
```

Comment:
```suggestion
    // If this function can start an unwind/rewind and is not the bottommost
```

Reply:
Is this right? Or did I misunderstand your intention?

But skimming the code below, I'm not sure how this relates to the bottommost runtime?

Reply:
So the current definition is `canChangeState` is true whenever a function can start unwinding or rewinding. But later, we set this to false when `isBottomMostRuntime` is true:
https://github.com/WebAssembly/binaryen/blob/0fe08192d866f72d0f4e680c862bb7af48dec1ac/src/passes/Asyncify.cpp#L629-L630

This was confusing to me so I added this. Let me know if I'm misunderstanding something.

Reply:
Oh, then maybe the better thing is to clarify with a comment there.

What's going on there is that we do not mark the bottom-most runtime as changing the state,

https://github.com/WebAssembly/binaryen/blob/0fe08192d866f72d0f4e680c862bb7af48dec1ac/src/passes/Asyncify.cpp#L515-L517

The top-most runtime can start a pause/resume, but the bottom is just where we unwind to and resume from. (That technically does modify the state, but we care about places we need to instrument for state changes, and the bottom runtime needs no such instrumentation - we don't need to "handle" those state changes like in user code.)

Reply:
I think it makes more sense to remove the change here and add a comment below on that operation.

But I don't feel strongly. I'd say though that if we want the comment right here then I'd replace it with something like this:

```diff
- If this function can start an unwind/rewind and not the bottommost runtime
+ If this function can start an unwind/rewind. We only set this in cases where we need to know that fact and also that we need to instrument code to handle it (as a result, we do not set it for the bottommost runtime, which needs no instrumentation).
```

Reply:
Thanks. Fixed as suggested.